ramips: add support for ASUS RT-AC1200GU

DragonBluep opened the pull request:
Specifications:
SoC:    MediaTek MT7621AT
FLASH:  16  MiB (Spansion S25FL128SA)
RAM:    128 MiB (Winbond W631GG6KB-15)
WiFi1:  MediaTek MT7603EN
WiFi2:  MediaTek MT7612EN
USB:    Type-A USB 2.0 *1
BTN:    Reset *1, WPS *1
LED:    Power *1, LAN *4, WAN *1, WiFi *2, USB *1

MAC Address:
label   xxxx:a5:10	factory@0xe006
wan     xxxx:a5:10	factory@0xe006
lan     xxxx:a5:14	factory@0xe000
wlan2g  xxxx:a5:10	factory@0x4
wlan5g  xxxx:a5:14	factory@0x8004

Installation:
Open SSH in the Web-interface and upload firmware to /tmp directory
via SCP, then execute instruction and reboot:
/# mtd-write -i /tmp/*.bin -d linux
/# reboot

Additional information:
This commit based on 14e0e4f138e3 ramips:add support for ASUS RT-AC57U

RT-AC57U and RT-AC1200GU are the same model sold in different countries.
It can be confirmed from the official support page that the two models
use the same firmware:
https://www.asus.com/Networking-IoT-Servers/WiFi-Routers/ASUS-WiFi-Routers/RT-AC57U/HelpDesk_Download/
https://www.asus.com/Networking-IoT-Servers/WiFi-Routers/ASUS-WiFi-Routers/RT-AC1200GU/HelpDesk_Download/

DragonBluep commented:
https://github.com/openwrt/openwrt/blob/c7727833945deb70eed7520ab44fa08b0870b8a7/target/linux/ramips/image/mt7621.mk#L166
I think there is a problem here because RT-AC57U/RT-AC1200GU has only one USB 2.0 port, should I correct it in this PR?

Borromini commented:
No issues there. I got the RT-AC57U, it only has USB 2.0 as well. The kmod-usb3 package handles USB 2 as well. Had that question myself back in the day, was told it's okay like this.

mrkiko commented:
Does this device support any recovery method?
Thanks!

DragonBluep commented:
> Does this device support any recovery method? Thanks!

You mean recovery to official firmware from OpenWrt, right? There are three Methods: 

1. Upload official firmware to /tmp via SCP, then execute instruction:
/# mtd -r write /tmp/*.trx firmware

2. Use "ASUS Firmware Restoration". Turn off power switch, Push reset button and hold on, then turn on power switch, after several seconds the router will enter recovery (the power led will blink) and you can loose your hand and upload firmware by "ASUS Firmware Restoration".

3. Maybe you can directly force apply update with official firmware in the LuCI interface (though I did not verify this way)

Everything above you can download from here: [ASUS Firmware Restoration and Firmware](https://www.asus.com/Networking-IoT-Servers/WiFi-Routers/ASUS-WiFi-Routers/RT-AC1200GU/HelpDesk_Download/)

mrkiko commented:
Thanks!!

chunkeey commented on target/linux/ramips/dts/mt7621_asus_rt-ac57u_ac1200gu.dtsi:
```diff
@@ -0,0 +1,186 @@
+	};
+};
+
+&factory {
```
this could be moved to the factory node definition above in line 85.

chunkeey commented on target/linux/ramips/dts/mt7621_asus_rt-ac57u_ac1200gu.dtsi:
```diff
@@ -0,0 +1,186 @@
+			gpios = <&gpio 48 GPIO_ACTIVE_LOW>;
+		};
+
+		usb {
```
The "label" properties have been deprecated upstream by a combination of [color + function + func-enum properties](https://www.kernel.org/doc/Documentation/devicetree/bindings/leds/common.yaml).

Note: we still need the label property for nodes which are assigned a led-* alias in the aliases node above.

A list of the possible color + function is available in:
https://elixir.bootlin.com/linux/v5.10.87/source/include/dt-bindings/leds/common.h#L24

mans0n commented:
If RT-AC57U and RT-AC1200GU are the exact same model, why don't you just use the image for already supported device? What benefit we gain from adding another files that only differs with model string?
(I once added a new device support that has the exact same hardware with already supported one, but that was because the new device bootloader demanded a different codename and I could not specify multiple initramfs image recipes in single device recipe.)
For listing aliases of the model, we can utilize `DEVICE_ALT*` variables.

DragonBluep commented:
> For listing aliases of the model, we can utilize `DEVICE_ALT*` variables.

I resubmitted it, thanks for your review.

mans0n commented:
Sorry, I feel the language in my last comment was a bit crude.

I checked the firmware of the two models and they indeed had the same md5sum.
Based on reports from wikidevi and articles I think it's safe to conclude RT-AC1200GU is a renaming of RT-AC57U and AFAIK `DEVICE_ALT*` was introduced for this kind of case.

And considering the previous examples adding alt device name, I think your commit message can be more concise.
ccf3bac6f3f9 08f5cac6fbd8 7b0156797609

DragonBluep committed:
ramips: add support for ASUS RT-AC1200GU

RT-AC57U and RT-AC1200GU are the same models sold in different countries.
The purpose of this commit is to allow users to easily find the
corresponding firmware through the model number on the device label.

More specifications: 14e0e4f138e3 "ramips:add support for ASUS RT-AC57U"

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

mans0n commented:
Thanks! Merged your patchset as 86d019d116c8.
